snuba/subscriptions/worker.py

```diff
@@ -47,3 +50,24 @@ def flush_batch(self, batch: Sequence[Sequence[SubscriptionResult]]) -> None:
         for results in batch:
             for result in results:
                 self.__producer.produce(self.__topic, result).result()
+
+
+class SubscriptionResultCodec(Codec[KafkaPayload, SubscriptionResult]):
+    def encode(self, value: SubscriptionResult) -> KafkaPayload:
+        subscription_id = str(value.task.task.identifier)
+        return KafkaPayload(
+            str(subscription_id).encode("utf-8"),
+            json.dumps(
+                {
+                    "version": 1,
+                    "payload": {
+                        "subscription_id": str(subscription_id),
```

Comment:
```suggestion
                        "subscription_id": subscription_id,
```

This is also redundant with the message key itself (which is accessible to the consumer) but I suppose it's not a big deal for it to be in both places.